### DISCUSSION TITLE
Table column values are overlapped after conversion from .wmf to .png

### DISCUSSION THREAD

**Discussion by lalitap25** [ImageMagick/ImageMagick#7590] [Help]
### ImageMagick version

7.1.1-Q16-HDRI

### Operating system, version and so on

Windows

### Description

Hi There,

I really need help with this matter.
I tried to convert some ole tables to png format 
But the converted image is not readable as the letters of the word inside column are overlapping. It is something to do with some setting so that I can view the image properly but I am not sure what settings to change

Here is the command that I ran on the imagemagick server

magick.exe convert C:\visualization\magick-tmp\picture6627189417406342295.wmf -density 600 -adaptive-resize 785 -gamma 0.9   C:\visualization\magick-tmp\picture6627189417406342295try10.png

The zip file contains there original file and the converted file.
[Images.zip](https://github.com/user-attachments/files/16860496/Images.zip)

**Comment by dlemstra** [comment]
You should not be using `magick.exe convert`. And you should put the commands in the correct order:

```
magick.exe -density 600 C:\visualization\magick-tmp\picture6627189417406342295.wmf -adaptive-resize 785 -gamma 0.9 C:\visualization\magick-tmp\picture6627189417406342295try10.png
```

**Reply by lalitap25** [reply]
Hi @dlemstra ,

The command that you provide did not work either. The result is same with overlapping words in the table column

**Reply by dlemstra** [reply]
It looks like your command should be translated to `magick picture6627189417406342295.wmf  -adaptive-resize 785 -gamma 0.9picture6627189417406342295try10.png`.  And I don't experience that overlapping issue with the latest release. Are you using the latest release?

**Reply by lalitap25** [reply]
I am using the latest release and its not working in my case.
<img width="606" alt="image" src="https://github.com/user-attachments/assets/f9632bed-4a61-44d0-a0d2-01d331f7f70e">

Do I need any additional dll file ?

**Reply by lalitap25** [reply]
Hi @dlemstra , 

Could you let me know which OS version you have  and which imagemagick you tried as in my case the latest release doesnt work.

**Reply by lalitap25** [reply]
I also checked directly on my Windows 10 laptop and the result is same. The issue is with the installer. You might have something additional on top of the installed files for the correct result

**Reply by dlemstra** [reply]
The emf coder uses Gdiplus to read the file. For some reason this behaves differently on your machine. I don't have and also don't want access to your machine so you will now need to figure out why this happens on your machine.

**Reply by dlemstra** [reply]
Have you tried upgrading the Windows SDK of your machine? It might be possible that you are on an older version that displays this incorrectly?

**Reply by lalitap25** [reply]
I checked, I do have the GDIplus.dll on my system. Also I tried to run the command in debug mode and no critical error comes up. If it was my system than there should have some error/warning for something missing. 
[imagemagicklog.txt](https://github.com/user-attachments/files/17384902/imagemagicklog.txt)

**Reply by dlemstra** [reply]
I also realized that you can also directly read the image in the size that you want without the resize. Do you get the correct output when you use `magick -size 785x picture6627189417406342295.wmf x.jpg`?

**Reply by dlemstra** [reply]
Those messages are from us and messages from GDIplus will not show up in that log file.

**Reply by dlemstra** [reply]
Looking at your log file it seems that you are getting different image dimensions from the GDI+ api. That is out of our control.

**Comment by snibgo** [comment]
I confirm the overlapping problem, using IM v7.1.1-38 downloaded today on Windows 11.
```
magick picture6627189417406342295.wmf x.jpg
```
x.jpg is:
![x](https://github.com/user-attachments/assets/c46826ec-04c8-44d0-ab61-f57c6403c53e)

**Comment by fmw42** [comment]
Perhaps check the date and/or version of GDIplus.dll and compare to what dlemstra is using

**Reply by dlemstra** [reply]
My system has version `10.0.26100.1882`.

**Reply by lalitap25** [reply]
Maybe thats why, in both my systems(windows 10 and windows server 2019) I dont have this version.
<img width="265" alt="windowserver2019gdiversion" src="https://github.com/user-attachments/assets/2e96a46b-6a5b-499f-a438-646f11445665">
<img width="272" alt="windows10gdiversion" src="https://github.com/user-attachments/assets/f3034201-f22a-4e51-8412-df9fab6db117">

**Reply by lalitap25** [reply]
Hi @snibgo ,

Is it possible for you to share this dll file with version 10.0.26100.1882

**Comment by snibgo** [comment]
> Hi @snibgo ,
Is it possible for you to share this dll file with version 10.0.26100.1882

I don't have that. I think you intended to ask @dlemstra.

**Reply by lalitap25** [reply]
Oh my bad. Hi @dlemstra , will it be possible for you to share the dll file?

**Reply by dlemstra** [reply]
You probably need to run some updates on your machine. I don't think it would be wise to just copy that library to your machine. It might have other dependencies that also need to be included.

**Comment by lalitap25** [comment]
Hi @dlemstra,

I upgraded my Windows SDK to the latest version but looks like it doesn't really change the dlls. It must be dependent on the OS(Ours is Windows server 2012/2019). If anyone has any other ideas , it will be really helpful as it is becoming a blocker for us.

**Reply by fmw42** [reply]
Check your font glyphs.  See https://usage.imagemagick.org/text/#font_info

**Comment by lalitap25** [comment]
Hi All,

I upgraded  my windows server to 2022 version and the issue still exist with it. The GDIplus.dll version is still older than what @dlemstra has. (10.0.20 compared to 10.0.26) Is it possible somehow you can provide the needed dll in the IM dll installer? I even tried to get the latest version of this dll independently and replaced it with the existing but looks like dlls doesn't work like that.

**Reply by dlemstra** [reply]
We will not going to add system files to our installer. It might fix it for you but it will be going to break it for someone else.
